pkg/kademlia/kademlia.go

```diff
@@ -377,6 +399,44 @@ func (k *Kad) ClosestPeer(addr swarm.Address) (swarm.Address, error) {
 	return closest, nil
 }
 
+// EachPeer iterates from closest bin to farthest
+func (k *Kad) EachPeer(f topology.EachPeerFunc) error {
+	return k.connectedPeers.EachBin(f)
+}
+
+// EachPeerRev iterates from farthest bin to closest
+func (k *Kad) EachPeerRev(f topology.EachPeerFunc) error {
+	return k.connectedPeers.EachBinRev(f)
+}
+
+// SubscribePeersChange returns the channel that signals when the connected peers
+// set changes. Returned function is safe to be called multiple times.
+func (k *Kad) SubscribePeersChange() (c <-chan struct{}, unsubscribe func()) {
+	channel := make(chan struct{}, 1)
+	var closeOnce sync.Once
+
+	k.peerSigMtx.Lock()
+	defer k.peerSigMtx.Unlock()
+
+	k.peerSig = append(k.peerSig, channel)
```

Comment:
🌷 Maybe it's a little bit nicer as the function below also locks, so someone might make a mistake while changing stuff.
```suggestion
	k.peerSigMtx.Lock()
	k.peerSig = append(k.peerSig, channel)
        k.peerSigMtx.Unlock()
```

Reply:
i don't think this is too confusing tbh and i prefer the `defer` statement